andela/travel_tool_front
# #160047860 Implement Table Filters on `My Approvals` Page

#### What does this PR do?
- Integrates the approvals table to the backend
  - Connects to redux state store
  - Creates actions related to fetching approvals
  - Creates an API for fetching a user's approvals with `axios`
- Implements table filters for the approvals page
- Enables pagination in the approvals table 

#### Description of Task to be completed?
##### Viewing approvals
When authenticated users need to fetch requests that they are required to approve, they should be able to see them in the approvals table. 

##### Filtering approvals
They should also be able to choose to view requests that are still pending their decision to approve or decline. If they decide to view their past decisions on different requests, they should be able to do that by hitting the `Past Approvals` filter which should present them with entries that they either individually approved or declined in the past i.e. independent of the decisions made in subsequent stages of the request approval process

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
- Change into repo directory - `cd travel_tool_front`
- Check out this branch - `git checkout ft-approvals-table-filters-160047860`
- Install dependencies and run the app - `make start`
- Navigate to `localhost:3000` in a browser
- Login and navigate to the requests page
- Click the Requests dropdown item on the left-side navigation bar
- Click `My Approvals` option and get you'll be presented with requests that the logged in user is required to approve
- Click on `Pending Approvals` and you should be presented with `Open` approvals only
- Click on `Past Approvals` and you should be presented with `Approved` and `Rejected` requests only
- Click on `Items per page` and you should be able to customize the number of entries you see on each page
  - Use the `next` and `previous` buttons at the bottom of the page to navigate the pages according to you liking

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?
#160047860

#### Screenshots (if appropriate)
<img width="1414" alt="screen shot 2018-09-06 at 4 55 42 pm" src="https://user-images.githubusercontent.com/23254512/45162040-c715d200-b1f5-11e8-83c8-ee2b6a9c4f0b.png">
<img width="1418" alt="screen shot 2018-09-06 at 4 55 50 pm" src="https://user-images.githubusercontent.com/23254512/45162041-c7ae6880-b1f5-11e8-829f-dd05e2fe17e5.png">
<img width="1420" alt="screen shot 2018-09-06 at 4 55 58 pm" src="https://user-images.githubusercontent.com/23254512/45162043-c7ae6880-b1f5-11e8-8cad-c46af5941452.png">


#### Questions:
